### PR TITLE
base64 image export to pdf

### PR DESCRIPTION
**Pull request by yogesharora388**
Changes made to export base64 image content to pdf

### CODE REVIEW

File: lib/src/html_to_widgets.dart [hunk 1]
```diff
@@ -14,6 +14,11 @@ import 'html_tags.dart';
 import 'pdfwidgets/bullet_list.dart';
 import 'pdfwidgets/number_list.dart';
 import 'pdfwidgets/quote_widget.dart';
+import 'dart:convert';
+
+RegExp base64RegExp = RegExp(
+  r'^(?:[A-Za-z0-9+/][A-Za-z0-9+/][A-Za-z0-9+/][A-Za-z0-9+/])*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$',
+);
```
Comment (alihassan143) [inline lib/src/html_to_widgets.dart:22]: why need to use regex? detection of base64 is easy

File: lib/src/html_to_widgets.dart [hunk 2]
```diff
@@ -549,9 +554,30 @@ class WidgetsHTMLDecoder {
     final src = element.attributes["src"];
     try {
       if (src != null) {
-        final netImage = await _saveImage(src);
-        return Image(MemoryImage(netImage),
-            alignment: customStyles.imageAlignment);
+        if (src.startsWith("data:image")) {
+          // To handle a case if someone added a space after base64 string
+          List components = src.split("base64, ");
```
Comment (alihassan143) [inline lib/src/html_to_widgets.dart:559]: why split with  base64, ?? you can use simple , to split becasue every base 64 start with that

File: lib/src/html_to_widgets.dart [hunk 3]
```diff
@@ -549,9 +554,30 @@ class WidgetsHTMLDecoder {
...
+          return Image(MemoryImage(listData),
+              alignment: customStyles.imageAlignment);
+        } else {
+          final netImage = await _saveImage(src);
+          return Image(MemoryImage(netImage),
+              alignment: customStyles.imageAlignment);
+        }
       } else {
         return Text("");
       }
```
Comment (alihassan143) [inline lib/src/html_to_widgets.dart:583]:
```
 try {
      if (src != null) {
        if (src.startsWith("data:image/")) {
          // To handle a case if someone added a space after base64 string
          final List<String> components = src.split(",");

          if (components.length > 1) {
            var base64Encoded = components.last;
            Uint8List listData = base64Decode(base64Encoded);
            return Image(MemoryImage(listData),
                alignment: customStyles.imageAlignment);
          }
          return Text("");
        }

        final netImage = await _saveImage(src);
        return Image(MemoryImage(netImage),
            alignment: customStyles.imageAlignment);
      }
      return Text("");
    } catch (e) {
      return Text("");
    }
```
that is more optimized code

**Review by alihassan143 — CHANGES_REQUESTED** [review]
kindly make changes and push code

File: lib/src/html_to_widgets.dart (hunk 1 above)
Comment (yogesharora388) [reply lib/src/html_to_widgets.dart:22]: I used it because if HTML does not contain prefix data:image/

**Review by n3o2k7i8ch5 — APPROVED** [review]

**Pull request merged**